### PR TITLE
Update build process and fix vulnerabilities

### PR DESCRIPTION
**Pull request by chris-armstrong**
Fixes issue #211

Changes: 
* Update index tests to start and stop server
* Upgrade package versions to avoid vulnerabilities (including dynamodb-localhost version)
* Add package-lock.json
* Add builds for node 8, 10 and 12
* Rewrite travis build for stability / updated node versions

Demo Link: N/A

Screenshots for the change: N/A

### CODE REVIEW

**Review by Toolito — APPROVED** [review]

**Comment by chanmathew** [comment]
Hi is there any update on the patch for this vulnerability?

image

**Comment by chris-armstrong** [comment]
I'm just waiting for the maintainers to review it - I don't really want to fork and start publishing a variant.

**Comment by piotr-cz** [comment]
Dependencies update also available in #231

**Review by mjzone — APPROVED** [review]

**Pull request merged**
